UnityProject/Assets/Scripts/Objects/VisibleBehaviour.cs

```diff
@@ -93,7 +95,16 @@ private void UpdateState(bool _aliveState)
 
 		for (int i = 0; i < renderers.Length; i++)
 		{
-			renderers[i].enabled = _aliveState;
+			SpriteRenderer sr = renderers[i] as SpriteRenderer;
+			// Cast and check cast.
+			// This is necessary because some renderers fail the cast.
+			if (sr != null && !CanBeDisabled(sr))

```

Comment:
Great this is clean code... Now test the heck out if it... 
(using two clients, one server, high ping and so on!)